### PR TITLE
Feature/theme output

### LINKED ISSUE #283: Output theming

### PR DESCRIPTION
**Pull request by FelixOpolka**
Closes #283 
Adds theme for lightstrip and seven-segment. Console will be done separately.

### CODE REVIEW

File: source/ui/Components/Output/ColorChooser.qml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import QtQuick.Controls 1.4
+import QtQuick.Dialogs 1.2
+
+ColorDialog {
+  id: colorDialog
+  width: 300
+  height: 200
+  title: "Please choose a color"
+
+  onRejected: {
```
Comment (goldsborough) [inline source/ui/Components/Output/ColorChooser.qml]: Same line

File: source/ui/Components/Output/ColorChooser.qml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+/*this modul is the contaier for the output-windows*/
+
+import QtQuick 2.6
+import QtQuick.Controls 1.4
+import QtQuick.Dialogs 1.2
+
+ColorDialog {
+  id: colorDialog
+  width: 300
```
Comment (goldsborough) [inline source/ui/Components/Output/ColorChooser.qml]: Theme

File: source/ui/Components/Output/Lightstrip/LightStrip.qml [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    // Send when any item's settings where updated.
+    onOutputItemSettingsChanged: {
+      updateContent(outputComponent.getOutputItem(outputItemIndex)["baseAddress"]);
+      lightstripSettingsWindow.updateSettings();
+    }
+  }
+
+  // Called from outside by the output tab view to signal that the settings button for the current
+  // output item was pressed.
+  onSettingsButtonPressed: {
```
Comment (goldsborough) [inline source/ui/Components/Output/Lightstrip/LightStrip.qml]: Same line

File: source/ui/Components/Output/Lightstrip/LightstripSettingsWindow.qml [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  flags: Qt.Dialog
+  modality: Qt.ApplicationModal
+
+  // Refreshes the window's control contentItem.
+  function updateSettings() {
+    numberOfStripsTextField.text = outputComponent.getOutputItem(outputItemIndex)["numberOfStrips"];
+    baseAddressTextField.text = "0x" + outputComponent.getOutputItem(outputItemIndex)["baseAddress"].toString(16);
+  }
+
+  onVisibleChanged: {
```
Comment (goldsborough) [inline source/ui/Components/Output/Lightstrip/LightstripSettingsWindow.qml]: Same line

File: source/ui/Components/Output/Lightstrip/LightstripSettingsWindow.qml [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    settingsWindow.updateSettings();
+  }
+
+  GridLayout {
+    id: grid
+
+    anchors.left: parent.left
+    anchors.leftMargin: 15
```
Comment (goldsborough) [inline source/ui/Components/Output/Lightstrip/LightstripSettingsWindow.qml]: Theme. There shouldn't be any numbers left in the QML at all

File: source/ui/Components/Output/Lightstrip/LightstripSettingsWindow.qml [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+    Text {
+      text: "Memory Source Address:"
+    }
+
+    // Text field for setting the output item's source address in memory.
+    TextField {
+      id: baseAddressTextField
+
+      onAccepted: { processInput(); }
```
Comment (goldsborough) [inline source/ui/Components/Output/Lightstrip/LightstripSettingsWindow.qml]: No need for braces

File: source/ui/Components/Output/Output.qml [hunk 7]
```diff
@@ -23,17 +23,20 @@
...
+Item {
     id: rootRectangle
 
     // Color definitions
-    property color tabBarColor: Qt.rgba(236.0/255.0, 236.0/255.0, 236.0/255.0, 1.0)
-    property color innerBorderColor: "#AFAFAF"
+    property color tabBarColor: Theme.output.tabBar.background
+    property color innerBorderColor: Theme.output.tabBar.border.color
     property color highlightColor: "#4A90E2"
     property color titleColor: "#4A4A4A"
```
Comment (goldsborough) [inline source/ui/Components/Output/Output.qml]: Theme

File: source/ui/Components/Output/SevenSegment/SevenSegment.qml [hunk 8]
```diff
@@ -28,144 +27,179 @@
...
+    id: sevenSegmentDigitsModel
+  }
+
+  // ListView for seven segment digits.
+  ListView {
+    id: digitsListView
+
+    anchors.centerIn: parent
+    height: parent.height - 10
+    spacing: 10
```
Comment (goldsborough) [inline source/ui/Components/Output/SevenSegment/SevenSegment.qml]: Theme

File: source/ui/Components/Output/SevenSegment/SevenSegmentSettingsWindow.qml [hunk 9]
```diff
@@ -18,154 +17,133 @@
...
+
+  onVisibleChanged: {
+    settingsWindow.updateSettings();
+  }
+
+  GridLayout {
+    id: grid
+
+    anchors.left: parent.left
+    anchors.leftMargin: 15
```
Comment (goldsborough) [inline source/ui/Components/Output/SevenSegment/SevenSegmentSettingsWindow.qml]: Theme

**Review by goldsborough — COMMENTED** [review]
Looks good. Have to test it in action

**Comment by goldsborough** [comment]
Ok looks nice. One thing: can we make the lightstrips smaller? I think they're too big. Better make them smaller and center them in the component.

**Review by goldsborough — APPROVED** [review]

**Comment by goldsborough** [comment]
`qrc:/Components/Output/Console/Console.qml:86: ReferenceError: it is not defined`

**Comment by FelixOpolka** [comment]
> qrc:/Components/Output/Console/Console.qml:86: ReferenceError: it is not defined

Changes to console are made on a separate branch.

**Pull request merged**